### PR TITLE
PATRON2020-145 Deployment Gateway.

### PR DESCRIPTION
**Pull request by SzymonFed**
EDIT: Renamed to Gateway deployment

### CODE REVIEW

File: gateway/gateway.cpp [hunk 1]
```diff
@@ -1,28 +1,145 @@
...
+#include "dashboard.h"
+#include "notifications.h"
 
-#define GATEWAY_PORT 42893
+#define GATEWAY_PORT 443
+#define CROW_ENABLE_SSL
 
-int main()
+#define CERT_FILE "/etc/letsencrypt/live/gate.patronage2020-iot.intive-projects.com/fullchain.pem"
+#define KEY_FILE "/etc/letsencrypt/live/gate.patronage2020-iot.intive-projects.com/privkey.pem"
```
Comment (Deelay1403) [inline gateway/gateway.cpp:10]: Wydaje mi się, że lokalizacja kluczy oraz tokeny powinny być w osobnym pliku dodanym do .gitignore

File: gateway/dashboard.h [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    {
+    _temperature_sensors = dashboard["temperatureSensors"].get<std::vector<TemperatureSensor>>();
+    _window_sensors = dashboard["windowSensors"].get<std::vector<WindowSensor>>();
+    _window_blinds = dashboard["windowBlinds"].get<std::vector<WindowBlind>>();
+    _rfid_sensors = dashboard["RFIDSensors"].get<std::vector<RFIDSensor>>();
+    _smoke_sensors = dashboard["smokeSensors"].get<std::vector<SmokeSensor>>();
+    _hvac_status = dashboard["HVACStatus"].get<HVACStatus>();
+    _hvac_rooms = dashboard["HVACRooms"].get<std::vector<HVACRoom>>();
+    _lights = dashboard["lights"].get<std::vector<Lights>>();
+    };*/
```
Comment (Siwy160) [inline gateway/dashboard.h:29]: There should be no commented code on development branch.

File: gateway/dashboard.h [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    void set_dashboard(json dashboard)
+    {
+    _temperature_sensors = dashboard["temperatureSensors"].get<std::vector<TemperatureSensor>>();
+    _window_sensors = dashboard["windowSensors"].get<std::vector<WindowSensor>>();
+    _window_blinds = dashboard["windowBlinds"].get<std::vector<WindowBlind>>();
+    _rfid_sensors = dashboard["RFIDSensors"].get<std::vector<RFIDSensor>>();
+    _smoke_sensors = dashboard["smokeSensors"].get<std::vector<SmokeSensor>>();
+    _hvac_status = dashboard["HVACStatus"].get<HVACStatus>();
+    _hvac_rooms = dashboard["HVACRooms"].get<std::vector<HVACRoom>>();
+    _lights = dashboard["lights"].get<std::vector<Lights>>();
```
Comment (Siwy160) [inline gateway/dashboard.h:93]: incorrect formatting here.

File: gateway/dashboard.h [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    int set_blind(json request)
+    {
+        if(request["type"] != "windowBlind")
+            return 1;
+        for(int i = 0;i<_window_blinds.size();i++)
```
Comment (Siwy160) [inline gateway/dashboard.h:100]: As discussed, it cannot be done that way. Would be good to leave proper comment with ticket corresponding ticket name.

File: gateway/dashboard.h [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                _lights[i]._hue = request["hue"].get<int>();
+                _lights[i]._saturation = request["saturation"].get<int>();
+                _lights[i]._value = request["value"].get<int>();
+                return 0;
+            }
+        }
+        return 1;
+    }
+    
+
```
Comment (Siwy160) [inline gateway/dashboard.h:146]: blank lines

File: gateway/gateway.cpp [hunk 6]
```diff
@@ -1,28 +1,145 @@
...
+        auto& cookiectx = lamGetCookieCtx(req);
+        if(cookiectx.get_cookie(COOKIE_NAME)!=COOKIE_VALUE)
+            res = crow::response(403);
+        else {CROW_LOG_INFO<<"COOKIE OK";}
+    }
+};
+
+int main()
+{
+
```
Comment (Siwy160) [inline gateway/gateway.cpp:71]: blank line

**Comment by Siwy160** [comment]
Also i don't understand all those merge commints. Would be good to squash them into single commit if possible.
It is rather minor. If assistance needed, please contact me via Slack.

File: gateway/dashboard.h (hunk 3 above)
Comment (SzymonFed) [reply gateway/dashboard.h]: Fixed

File: gateway/gateway.cpp (hunk 6 above)
Comment (SzymonFed) [reply gateway/gateway.cpp:75]: Fixed

File: gateway/dashboard.h (hunk 2 above)
Comment (SzymonFed) [reply gateway/dashboard.h]: Removed

**Comment by SzymonFed** [comment]
Squashed unnecessary merge commits. They were an effect of using github to move binary file into server (:D).

File: gateway/dashboard.h (hunk 4 above)
Comment (SzymonFed) [reply gateway/dashboard.h]: I don't remember why I can't do this that way. Could you elaborate on this?

File: gateway/gateway.cpp (hunk 1 above)
Comment (SzymonFed) [reply gateway/gateway.cpp:14]: Zastanawiam się jak to ma wyglądać. Czy:
- umieścić to w osobnym nagłówku, który poprostu nie będzie wrzucony na gita?
- zrobić jakis config file ktory będzie parsowany?

File: gateway/gateway.cpp [hunk 7]
```diff
@@ -1,28 +1,149 @@
 #include "dependencies/crow_all.h"
+#include "dependencies/json.hpp"
+#include "dashboard.h"
+#include "notifications.h"
 
+<<<<<<< HEAD
+#define GATEWAY_PORT 443
+=======
 #define GATEWAY_PORT 42893
+>>>>>>> 8efe5bc... Added cookie check, added ssl[untested]
```
Comment (cichy1024) [inline gateway/gateway.cpp:10]: Did you commit an unmerged file?

File: gateway/sensors.h [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+enum window_status
+{
+    OPEN,
+    CLOSED
+};
+
+NLOHMANN_JSON_SERIALIZE_ENUM(window_status,{
+    {OPEN,"open"},
+    {CLOSED,"closed"}
+
```
Comment (cichy1024) [inline gateway/sensors.h:14]: Please remove this blank line and similar ones.

File: gateway/gateway.cpp (hunk 1 above)
Comment (cichy1024) [reply gateway/gateway.cpp:14]: I would leave it as is for the moment. We can create another ticket for that. Let's make it working somehow right now.

File: gateway/gateway.cpp (hunk 7 above)
Comment (SzymonFed) [reply gateway/gateway.cpp]: I don't know why this happened. Should be fixed now.

File: gateway/sensors.h (hunk 8 above)
Comment (SzymonFed) [reply gateway/sensors.h]: Fixed

**Pull request merged**